File: 2025-09-25 - ListaPrioridade/Exemplo_Java/Node.java
// ------------------------------------------------------------------------
// Node.java
// Classe que representa um nó da lista de prioridade
// Cada nó possui código, nome, carga horária e semestre (que define a prioridade)
// ------------------------------------------------------------------------
public class Node {
   // ---------------------------------------------------------------------
   // Atributos privados
   // ---------------------------------------------------------------------
   private String codigoDisciplina;
   private String nomeDisciplina;
   private int cargaHoraria;
   private int semestre;
   private Node proximo; // referência para o próximo nó

   // ---------------------------------------------------------------------
   // Construtor
   // ---------------------------------------------------------------------
   public Node(String codigoDisciplina, String nomeDisciplina, int cargaHoraria, int semestre) {
      this.codigoDisciplina = codigoDisciplina;
      this.nomeDisciplina = nomeDisciplina;
      this.cargaHoraria = cargaHoraria;
      this.semestre = semestre;
      this.proximo = null;
   }

   // ---------------------------------------------------------------------
   // Getters e Setters
   // ---------------------------------------------------------------------
   public String getCodigoDisciplina() {
      return codigoDisciplina;
   }

   public void setCodigoDisciplina(String codigoDisciplina) {
      this.codigoDisciplina = codigoDisciplina;
   }

   public String getNomeDisciplina() {
      return nomeDisciplina;
   }

   public void setNomeDisciplina(String nomeDisciplina) {
      this.nomeDisciplina = nomeDisciplina;
   }

   public int getCargaHoraria() {
      return cargaHoraria;
   }

   public void setCargaHoraria(int cargaHoraria) {
      this.cargaHoraria = cargaHoraria;
   }

   public int getSemestre() {
      return semestre;
   }

   public void setSemestre(int semestre) {
      this.semestre = semestre;
   }

   public Node getProximo() {
      return proximo;
   }

   public void setProximo(Node proximo) {
      this.proximo = proximo;
   }
}

File: 2025-09-25 - ListaPrioridade/Exemplo_Python/Node.py
# ------------------------------------------------------------------------
# Node.py
# Classe que representa um nó da lista de prioridade
# ------------------------------------------------------------------------
class Node:
    """
    Cada nó possui código, nome, carga horária e semestre 
    (que define a prioridade).
    """
    def __init__(self, codigoDisciplina, nomeDisciplina, cargaHoraria, semestre):
        self._codigoDisciplina = codigoDisciplina
        self._nomeDisciplina = nomeDisciplina
        self._cargaHoraria = cargaHoraria
        self._semestre = semestre
        self._proximo = None

    # --- Getters ---
    def getCodigoDisciplina(self):
        return self._codigoDisciplina

    def getNomeDisciplina(self):
        return self._nomeDisciplina

    def getCargaHoraria(self):
        return self._cargaHoraria

    def getSemestre(self):
        return self._semestre

    def getProximo(self):
        return self._proximo

    # --- Setters ---
    def setCodigoDisciplina(self, codigoDisciplina):
        self._codigoDisciplina = codigoDisciplina

    def setNomeDisciplina(self, nomeDisciplina):
        self._nomeDisciplina = nomeDisciplina

    def setCargaHoraria(self, cargaHoraria):
        self._cargaHoraria = cargaHoraria

    def setSemestre(self, semestre):
        self._semestre = semestre

    def setProximo(self, proximo):
        self._proximo = proximo